src/engine/Operation.cpp

```diff
@@ -211,144 +209,95 @@ void Operation::checkTimeout() const {
 // _______________________________________________________________________
 void Operation::updateRuntimeInformationOnSuccess(
     const ResultTable& resultTable, ad_utility::CacheStatus cacheStatus,
-    size_t timeInMilliseconds, std::optional<RuntimeInformation> runtimeInfo) {
-  _runtimeInfo.totalTime_ = timeInMilliseconds;
-  _runtimeInfo.numRows_ = resultTable.size();
-  _runtimeInfo.cacheStatus_ = cacheStatus;
+    size_t timeInMilliseconds) {
+  _runtimeInfo->totalTime_ = timeInMilliseconds;
+  _runtimeInfo->numRows_ = resultTable.size();
+  _runtimeInfo->cacheStatus_ = cacheStatus;
 
-  _runtimeInfo.status_ = RuntimeInformation::Status::fullyMaterialized;
+  _runtimeInfo->status_ = RuntimeInformation::Status::fullyMaterialized;
 
   bool wasCached = cacheStatus != ad_utility::CacheStatus::computed;
-  // If the result was read from the cache, then we need the additional
-  // runtime info for the correct child information etc.
-  AD_CONTRACT_CHECK(!wasCached || runtimeInfo.has_value());
-
-  if (runtimeInfo.has_value()) {
-    if (wasCached) {
-      _runtimeInfo.originalTotalTime_ = runtimeInfo->totalTime_;
-      _runtimeInfo.originalOperationTime_ = runtimeInfo->getOperationTime();
-      _runtimeInfo.details_ = std::move(runtimeInfo->details_);
-    }
-    // Only the result that was actually computed (or read from cache) knows
-    // the correct information about the children computations.
-    _runtimeInfo.children_ = std::move(runtimeInfo->children_);
-  } else {
-    // The result was computed by this operation (not read from the cache).
-    // Therefore, for each child of this operation the correct runtime is
-    // available.
-    _runtimeInfo.children_.clear();
-    for (auto* child : getChildren()) {
-      AD_CONTRACT_CHECK(child);
-      _runtimeInfo.children_.push_back(
-          child->getRootOperation()->getRuntimeInfo());
-    }
+
+  if (wasCached) {
+    _runtimeInfo->originalOperationTime_ = _runtimeInfo->getOperationTime();
   }
```

Comment:
I think this is wrong,
When we read from the cache, we also have to inherit the results from the children. 
Also this means that "our" version of these children never were computed, so there is nothing left to remember there.

Reply:
So actually the old code can almost stay as is, maybe with some adjustments to the handling of shared_ptrs.